src/rust/scheduler/scheduler.rs

```diff
@@ -137,108 +137,106 @@ impl Scheduler {
         let mut tasks: RefMut<PinSlab<Box<dyn Task>>> = self.tasks.borrow_mut();
 
         // Iterate through pages.
-        for page_ix in 0..pages.len() {
-            let (notified, dropped): (u64, u64) = {
-                let page: &mut WakerPageRef = &mut pages[page_ix];
-                (page.take_notified(), page.take_dropped())
+        for page_index in 0..page_refs.len() {
+            let (notified_offsets, dropped_offsets): (u64, u64) = {
+                let page_ref: &mut WakerPageRef = &mut page_refs[page_index];
+                (page_ref.take_notified(), page_ref.take_dropped())
             };
             // There is some notified task in this page, so iterate through it.
-            if notified != 0 {
-                for subpage_ix in BitIter::from(notified) {
-                    // Handle notified tasks only.
-                    // Get future using our page indices and poll it!
-                    let ix: usize = (page_ix << WAKER_BIT_LENGTH_SHIFT) + subpage_ix;
+            if notified_offsets != 0 {
+                for page_offset in BitIter::from(notified_offsets) {
+                    // Get future using page indices and poll it!
+                    let pin_slab_index: usize = (page_index << WAKER_BIT_LENGTH_SHIFT) + page_offset;
                     let waker: Waker = unsafe {
-                        let raw_waker: NonNull<u8> = pages[page_ix].into_raw_waker_ref(subpage_ix);
+                        let raw_waker: NonNull<u8> = page_refs[page_index].into_raw_waker_ref(page_offset);
                         Waker::from_raw(WakerRef::new(raw_waker).into())
                     };
                     let mut sub_ctx: Context = Context::from_waker(&waker);
 
-                    let pinned_ref: Pin<&mut Box<dyn Task>> = tasks.get_pin_mut(ix).unwrap();
+                    let pinned_ref: Pin<&mut Box<dyn Task>> = tasks.get_pin_mut(pin_slab_index).unwrap();
                     let pinned_ptr = unsafe { Pin::into_inner_unchecked(pinned_ref) as *mut _ };
 
                     // Poll future.
-                    drop(pages);
+                    drop(page_refs);
                     drop(tasks);
                     let pinned_ref = unsafe { Pin::new_unchecked(&mut *pinned_ptr) };
                     let poll_result: Poll<()> = Future::poll(pinned_ref, &mut sub_ctx);
-                    pages = self.pages.borrow_mut();
+                    page_refs = self.page_refs.borrow_mut();
                     tasks = self.tasks.borrow_mut();
                     match poll_result {
-                        Poll::Ready(()) => pages[page_ix].mark_completed(subpage_ix),
+                        Poll::Ready(()) => page_refs[page_index].mark_completed(page_offset),
                         Poll::Pending => (),
                     }
                 }
             }
             // There is some dropped task in this page, so iterate through it.
-            if dropped != 0 {
-                // Handle dropped tasks only.
```

Comment:
Please, do not remove existing comments.

Reply:
I was mislead by the comment to think that there would be a check whether the task is dropped or not. But, this code only iterates over the `dropped` bits, so there is no need of a check.

Reply:
The comment does not say "check for dropped tasks", it says to "handle" them. If the comment is not good as expressive as it could be, let's update it, not remove.

Reply:
The wording in this comment (the word only) led me to think that there would be a check, so I went searching for it. If we are changing the existing comment becuase it is not helpful, then removing it altogether makes sense. Do let me know if you have any suggestions for what needs to be commented at this place - especially when the comment 2 lines above explains what the code below is going to do.

Reply:
I think we do not need to discuss this because we want to get rid of this. In fact if there is some way to mark the code as deprecated, that would be great.

Reply:
Ok, I reverted the comment. We can discuss offline about more on deprecation.